src/utils/viewer/viewer.cpp

```diff
@@ -552,11 +588,90 @@ void Viewer::viewportEvent(ViewportEvent& event) {
   renderCamera_->setViewport(event.windowSize());
   imgui_.relayout(Mn::Vector2{event.windowSize()} / event.dpiScaling(),
                   event.windowSize(), event.framebufferSize());
+
+  /* Recreate object ID reading renderbuffers that depend on viewport size */
+  selectionDepth_ = Mn::GL::Renderbuffer{};
+  selectionDepth_.setStorage(Mn::GL::RenderbufferFormat::DepthComponent24,
+                             event.framebufferSize());
+  selectionDrawableId_ = Mn::GL::Renderbuffer{};
+  selectionDrawableId_.setStorage(Mn::GL::RenderbufferFormat::R32UI,
+                                  event.framebufferSize());
+  selectionFramebuffer_
+      .attachRenderbuffer(Mn::GL::Framebuffer::BufferAttachment::Depth,
+                          selectionDepth_)
+      .attachRenderbuffer(Mn::GL::Framebuffer::ColorAttachment{1},
+                          selectionDrawableId_)
+      .setViewport({{}, event.framebufferSize()});
 }
 
 void Viewer::mousePressEvent(MouseEvent& event) {
+  if (event.button() == MouseEvent::Button::Right && objectSelectionOn_) {
```

Comment:
This is a cool feature and there's a fair amount of new code in viewer.cpp to support it. It'd be nice to see all this code encapsulated in an ObjectPickerHelper class which the viewer uses.